### ISSUE TITLE
add format-string

### ISSUE THREAD

**Issue by sils1297** [coala-analyzer/coala#831]
hey, would be pretty cool to let the user enter a format string and we just print out all results formatted with that string. Would be really easy actually:

`coala --format="{file}|{message}|{debug_data}"` and we could just write a `print_results` callback like:

```python
lock = Lock()
format_str = "{file}|{message}|{debug_msg}"
def print_results(results):
    with lock:
        for result in results:
            print(format_str.format(**(result.__dict__)))
```

**Comment by sils1297** [comment]
but before we're doing that we'll want to minify our main methods ;)

**Comment by AbdealiJK** [comment]
This is actually not blocked by reducing the main methods - because we'd define this in `Interactions.py` and just use it in the main method.

Question is how do we do this ? Do we make another `coala-format` executable ? because I notice that the number of executables is increasing pretty fast ...

**Comment by sils1297** [comment]
@AbdealiJK agreed, IMO it's nice to have so many executables, I also thought about making something similar like git: the `git` executable simply takes `git <subcommand>` and invokes the `git-subcommand` executable :)

OTOH I can see how JSON and formatted output and maybe even coala-ci belong into the coala cli so we might consider adding params for that.
